packages/@webex/contact-center/test/unit/spec/services/core/Utils.ts

```diff
@@ -326,4 +326,346 @@ describe('Utils', () => {
+        },
+      };
+
+      const result = Utils.getConsultedAgentId(media, currentAgentId);
+      expect(result).toBe('');
+    });
+
+    it('should handle empty media object', () => {
+      const result = Utils.getConsultedAgentId({}, currentAgentId);
+      expect(result).toBe('');
+    });
+
+    it('should handle multiple media entries and find consult', () => {
+      const media: any = {
+        media1: {mType: 'mainCall', participants: [currentAgentId]},
+        media2: {mType: 'hold', participants: []},
+        media3: {mType: 'consult', participants: [currentAgentId, 'consulted-agent']},
+      };
+
+      const result = Utils.getConsultedAgentId(media, currentAgentId);
+      expect(result).toBe('consulted-agent');
+    });
+  });
+
+  describe('getDestAgentIdForCBT', () => {
+    it('should return destination agent ID for CBT scenario', () => {
+      const interaction: any = {
+        participants: {
+          'agent-uuid-123': {
+            type: 'Agent',
+            pType: 'dn',
+            dn: '5551234567',
+            id: 'agent-uuid-123',
+          },
+          'customer-1': {
+            type: 'Customer',
+            pType: 'Customer',
+            id: 'customer-1',
+          },
+        },
+      };
+      const consultingAgent = '5551234567'; // Phone number, not in participants as key
+
+      const result = Utils.getDestAgentIdForCBT(interaction, consultingAgent);
+      expect(result).toBe('agent-uuid-123');
+    });
+
+    it('should return empty string when consultingAgent is in participants (non-CBT)', () => {
+      const interaction: any = {
+        participants: {
+          'agent-123': {
+            type: 'Agent',
+            pType: 'Agent',
+            id: 'agent-123',
+          },
+        },
+      };
+      const consultingAgent = 'agent-123'; // Exists as key in participants
+
+      const result = Utils.getDestAgentIdForCBT(interaction, consultingAgent);
+      expect(result).toBe('');
+    });
+
+    it('should return empty string when no matching DN found', () => {
+      const interaction: any = {
+        participants: {
+          'agent-uuid-123': {
+            type: 'Agent',
+            pType: 'dn',
+            dn: '5559999999',
+            id: 'agent-uuid-123',
+          },
+        },
+      };
+      const consultingAgent = '5551234567'; // Different number
+
+      const result = Utils.getDestAgentIdForCBT(interaction, consultingAgent);
+      expect(result).toBe('');
+    });
+
+    it('should return empty string when consultingAgent is empty', () => {
+      const interaction: any = {
+        participants: {
+          'agent-uuid-123': {
+            type: 'Agent',
+            pType: 'dn',
+            dn: '5551234567',
+          },
+        },
+      };
+
+      const result = Utils.getDestAgentIdForCBT(interaction, '');
+      expect(result).toBe('');
+    });
+
+    it('should match only when pType is dn and type is Agent', () => {
```

Comment:
nit: Kindly adjust this everywhere.

```suggestion
    it('should match only when participant type is dial number and type is Agent', () => {
```

Reply:
Addressed